src/main/webapp/app/lecture/lecture.component.ts

```diff
@@ -117,4 +132,37 @@ export class LectureComponent implements OnInit, OnDestroy {
             error: (error: HttpErrorResponse) => this.dialogErrorSource.next(error.message),
         });
     }
+
+    /**
+     * Filters Lectures
+     * @param filterChecked the filter checkbox that was clicked
+     */
+    applyFilters(): void {
+        // Get the current system time
+        const currentTime = dayjs();
+        // Initialize empty arrays for filtered Lectures
+        let filteredLectures: Array<Lecture> = [];
+
+        // update filteredLectures based on the selected filter option checkboxes
+        const pastLectures = this.lectures.filter((lecture) => lecture.startDate !== undefined && lecture.endDate?.isBefore(dayjs(currentTime)));
```

Comment:
Why do we have to have a start date, if the end date is before today?

Reply:
I think the idea is that lectures that do not have a start _and_ end date belong to the _unspecified date_ group.